be/src/exec/tablet_info.cpp

```diff
@@ -66,30 +68,31 @@ Status OlapTableSchemaParam::init(const POlapTableSchemaParam& pschema) {
     for (auto& col : pschema.partial_update_input_columns()) {
         _partial_update_input_columns.insert(col);
     }
-    std::map<std::string, SlotDescriptor*> slots_map;
+    std::unordered_map<std::pair<std::string, std::string>, SlotDescriptor*> slots_map;
     _tuple_desc = _obj_pool.add(new TupleDescriptor(pschema.tuple_desc()));
 
     for (auto& p_slot_desc : pschema.slot_descs()) {
         auto slot_desc = _obj_pool.add(new SlotDescriptor(p_slot_desc));
         _tuple_desc->add_slot(slot_desc);
-        slots_map.emplace(slot_desc->col_name(), slot_desc);
+        string data_type;
+        EnumToString(TPrimitiveType, to_thrift(slot_desc->col_type()), data_type);
```

Comment:
we should check we get a type here, otherwise an unexpected behavior happens.

Reply:
When we init the tablet column, the field `column_type` will change from TPrimitiveType to string by this method to FieldType by `TabletColumn::get_field_type_by_string`. And the column type in columnPB is string and it changed from TPrimitiveType.